ASPNetCoreMastersTodoList.Api/Filters/EnsureItemIdExistFilter.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ASPNetCoreMastersTodoList.Api.Filters
+{
+    public class EnsureItemIdExistFilter : IActionFilter
+    {
+        private readonly IItemService _service;
+        public EnsureItemIdExistFilter(IItemService service)
+        {
+            _service = service;
+        }
+        public void OnActionExecuted(ActionExecutedContext context) {}
+
+        public void OnActionExecuting(ActionExecutingContext context)
+        {
+            var itemId = (int)context.ActionArguments["itemId"];
+            var isExist = _service.Get(itemId);
+            if (isExist == null)
```

Comment:
please handle scenarios where the action doesn't need to ensure an item exists (no itemId parameter)